src/ship/PrecalcPath.cpp

```diff
@@ -0,0 +1,551 @@
+// Copyright © 2008-2020 Pioneer Developers. See AUTHORS.txt for details
+// Licensed under the terms of the GPL v3. See licenses/GPL-3.txt
+
+#include "PrecalcPath.h"
+#include "utils.h"
+
+#include <algorithm>
+#include <cmath>
+
+// path element calculation functions
+// all information for the calculation is taken only from the arguments
+//
+// notation:
+//
+//   units:
+//     m:    mass, kg
+//     F:    forward thruster power, N
+//     EV:   effective exhaust velocity, m/s
+//     t:    time, s
+//     S:    path length, m
+//     acap: acceleration limit, m/s^2
+//     V:    ship velocity, m/s
+//     d<X>: change in variable <X> (dm, dV, dt ...)
+//
+//   function names postfixes:
+//     ..._norm  - normal mode, F = max, dm/dt = const, a = f(t)
+//     ..._acap  - limited mode, F = f(t), dm/dt = f(t), a = const = acap
+//     ..._mixed - use normal mode until F/m hits acap, then limited acceleration mode
+//
+//   stadia:
+//     1 - acceleration
+//     2 - decceleration
+//     i.e. t1 - acceleration time, S1 - acceleration path, m1 - mass after acceleration e.t.c.
+//     since free flight is very trivial it is not numbered
+//     if no stage is specified, the calculation is the same for both stages
+//
+//   example:
+//     S1_from_t1_norm() - calculate acceleration path from acceleration time, with no acceleration limit
+
+using val = const double &;
```

Comment:
I'm not sure why you would need to use `const double &` here - double is a scalar value type that is the same size as the implicit pointer that makes up a pass-by-reference function argument.

Were this being compiled in unoptimized mode, it would require extra instruction(s) to take a reference to a value and pass it to a function rather than just copying the double on the stack. When this code compiled in optimized mode, all of these functions are getting inlined anyways; which means there's no actual benefit to this over typedef-ing this to just `double`.

Reply:
Done!